# Project Title: YouTube Keyword Prediction for Trending Videos


### Loading Nigeria's Current Trending YouTube Videos:

###### Importing packages and reading data

In [8]:
# Importing important Libraries
import pandas as pd
import numpy as np

In [9]:
#importing the NigeriaYoutube data 
Ng_data= pd.read_csv('NG_youtube_data.csv')

In [7]:
Ng_data.head(4)

title      videoId  \
0  FAMILY MATTERS | Brodashaggi | Mr Ibu | Kiriku...  Zv1i-Cdn4Ik   
1  Best Friends in the World: Senior Year | Episo...  BmCkK7W7j0Q   
2  ROMOLA ALAKARA - Latest 2022 Yoruba Movie Star...  seZb9P7H6MU   
3                                       Money & Love  pbATe_uZFTE   

                  channelId           channelTitle  categoryId  \
0  UCG6orNVuXIICv9_ifH6msIA  official BRODA SHAGGI          23   
1  UCCl71WmqkG8BjVuySOSCh_Q       Neptune3 Studios           1   
2  UCK2-495cMvdhaMqXcLlgmcg             YORUBAPLAY          24   
3  UC25DGtgYk5TwdksvtpeWYvA         Wizkid - Topic          10   

  VideoCategoryDescription                                        Description  \
0                   Comedy  Please SUBSCRIBE to Broda Shaggi Official YouT...   
1         Film & Animation  EP 13 "The Protest" - Olive leads a protest ag...   
2            Entertainment  Find out where the story leads in this latest ...   
3                    Music  Provided to YouTube by Starboy/RCA Records\n\n...   

   viewCount  likeCount  dislikeCount  commentCount  favoriteCount  \
0     654033      30422             0          1474              0   
1     299169      21292             0          2012              0   
2     553886       5252             0           586              0   
3     197492       8561             0             0              0   

  publishedAt   duration  
0  22/10/2022   PT10M18S  
1  28/10/2022  PT1H5M22S  
2  21/10/2022  PT1H9M27S  
3  27/10/2022    PT3M12S

In [6]:
NigeriaYoutubeData.tail()

title      videoId  \
45  Phyno - BBO (Bad Bxtches Only) [Official Video]  TCchFx_dgp8   
46                 The Rock Vs MrBeast For $100,000  S_CUEOBZ0P4   
47                           Akpan and Oduma 'JAPA'  2C1DTt1JFUY   
48                Kizz daniel BUGA Wedding Entrance  oOzKbiWWl20   
49                                Police Internship  veIB8gLx5GM   

                   channelId       channelTitle  categoryId  \
45  UC_kzqsFmGmwcJPqEuGqUyGA          PhynoFino          10   
46  UCX6OQ3DkcsbYNE6H8uQQuVA            MrBeast          24   
47  UCtl7BVgRMGDUcLOD1MIqiYw       wapTVchannel          24   
48  UCQHJ0bJK5Z7m7EGbBoU-WDA  K.K.T Productions          22   
49  UC26q7KQxr5TnKm-OT25MyyQ       HouseOfAjebo          23   

   VideoCategoryDescription  \
45                    Music   
46            Entertainment   
47            Entertainment   
48           People & Blogs   
49                   Comedy   

                                          Description  viewCount  likeCount  \
45  BBO - Bad Bxtches Only\n\nStream on all platfo...     139417       6236   
46  Winning totally Rocks!\n\nNew Merch - https://...   34543612    3757194   
47  Finally, Akpan lives Oduma. Is he living for g...     244599      10059   
48                                                NaN     574391      10640   
49  Tegwolo's father has just joined the police. T...     135198       5985   

    dislikeCount  commentCount  favoriteCount publishedAt  duration  
45             0           502              0  27/10/2022   PT2M46S  
46             0         25955              0  21/10/2022     PT55S  
47             0           598              0  24/10/2022  PT16M25S  
48             0            46              0  16/10/2022     PT34S  
49             0           195              0  22/10/2022   PT5M49S

### Loading United State's Current Trending YouTubeVideos: 
###### Importing packages and reading data

In [13]:
US_data = pd.read_csv('us_Youtube_data.csv')

In [14]:
US_data.head()

channelId           channelTitle      videoId  \
0  UCU1_l0ZJyTK_7HZZ3Ruw8Dg                   MAPS  pTnk3ziVVRM   
1  UCLuO2lUqHrPIIpx0hFenV2g         Tink Tink Club  cuJjSeHZIrg   
2  UCihqrkaOgVMfLNo2W1hSliA           Podcast Bunk  IuyuZfWtGgg   
3  UCgbWWPn3VYYzxjffZbfj9GQ        Alan Springwind  cng_ZhQf8iY   
4  UCFmLi6X1mojkFZOFngNR9tQ  Drug Education Agency  OpQIQEx7J5A   

                publishedAt  \
0  2014-01-10T01:24:57.000Z   
1  2015-06-18T16:56:04.000Z   
2  2016-05-01T05:33:13.000Z   
3  2016-01-25T04:48:22.000Z   
4  2014-08-15T10:53:58.000Z   

                                          videoTitle  \
0  Psychedelic Horizons Beyond Psychotherapy Work...   
1                     Episode 35 - Dr. James Fadiman   
2  #325 Microdosing from The Adam and Dr Drew Sho...   
3                Microdosing Away The Same Old Blues   
4  Erschossener Kiffer / Drogen in Mikro-Dosierun...   

                                    videoDescription  videoCategoryId  \
0  Watch the full workshop at http://psychedelics...               29   
1  Dr. James Fadiman is the father of modern psyc...               22   
2  Adam and Dr. Drew are solo today and they open...               22   
3  Source: https://www.spreaker.com/user/springwi...               24   
4  Von erschossenen "Dealern", von demonstrierend...               24   

      videoCategoryLabel    duration  durationSec definition  caption  \
0  Nonprofits & Activism  PT1H19M40S         4780         hd    False   
1         People & Blogs  PT1H12M34S         4354         sd    False   
2         People & Blogs    PT50M40S         3040         sd    False   
3          Entertainment    PT32M21S         1941         hd    False   
4          Entertainment    PT26M32S         1592         hd    False   

   viewCount  likeCount  dislikeCount  commentCount  
0       1512        8.0           0.0           1.0  
1        881       14.0           0.0           3.0  
2         67        0.0           1.0           3.0  
3          4        0.0           0.0           0.0  
4       5465      121.0           6.0          25.0

In [13]:
USYoutubeData.tail()

Unnamed: 0                 channelId          channelTitle      videoId  \
110         110  UCjnXuO6mfE0-czvcbOYnpbQ       Ayahuasca Magic  j9HFlDff2XY   
111         111  UC0dsjSDsLZ5hbmoNjvsHJsQ         fluidresearch  lOGAkCtT4wk   
112         112  UCmNZB3-zCyMuHCOTraws0dg  Helix Steel Victoria  sTNaGjt6564   
113         113  UC281yIVp-U8Ijg4KUVRgDRA              Osiron X  fMlQoGINGHo   
114         114  UCH4ezCqRXHQhsETfl0dainA     FabFormIndustries  ODX4tzCWhhU   

                  publishedAt  \
110  2016-03-25T04:58:34.000Z   
111  2010-08-20T14:46:56.000Z   
112  2014-05-19T12:14:36.000Z   
113  2016-04-15T19:31:48.000Z   
114  2015-08-19T18:25:20.000Z   

                                            videoTitle  \
110  Time differentials: mystery wisdom and microdo...   
111  Fluid filling with exact precision dosing by F...   
112  Accurate dosing of Helix Twisted Steel Micro R...   
113                   Psilocybin Mushroom Micro-Dosing   
114                           Dosing Helix Micro Rebar   

                                      videoDescription  videoCategoryId  \
110  The lectures I talk about are free (some of th...               24   
111  See how the industry leaders in Precision micr...               26   
112  © Copyright --- Look at the steady rain of Hel...               28   
113  When too much time lapses between trips, the e...               24   
114  Coquitlam concrete goes into detail how they d...               27   

       videoCategoryLabel duration  durationSec definition  caption  \
110         Entertainment    PT26M           26         hd    False   
111         Howto & Style    PT18S           18         sd    False   
112  Science & Technology    PT11S           11         hd    False   
113         Entertainment    PT11M           11         sd    False   
114             Education     PT1M            1         hd    False   

     viewCount  likeCount  dislikeCount  commentCount  
110        304        6.0           0.0          22.0  
111        294        0.0           0.0           0.0  
112        463        0.0           0.0           0.0  
113        341       11.0           1.0          10.0  
114        345        4.0           0.0           0.0

### Sorting Needed Columns For Nigeria's Current Trending YouTube Video Dataset:
###### Here we are creating a dataframe that shows the trending Nigerian Dataset by Video Category Decription and View Counts in ascending order.

###### So we have 7 trending Nigerian YouTube Video Categories and they are as follows:
       Comedy, Entertainment, Film & Animation, Music, News & Politics, People & Blogs, and Sports.
    
###### The highest trending video under Comedy has 1,629,458 viewCounts
###### The highest trending video under <font color=red>Entertainment</font> has 34,543,612 viewCounts
###### The highest trending video under Film & Animation has 1,648,620 viewCounts
###### The highest trending video under Music has 12,173,797 viewCounts 
###### The highest trending video under News & Politics has 274,827 viewCounts
###### The highest trending video under People & Blogs has 574,391 viewCounts
###### The highest trending video under Sports has 7,154,769 viewCounts

###### Out of all the 7 trending Nigerian Youtube Video Categories, <font color=red>Entertainment</font> seems to have the most viewcount , therefore making it the highest trending youtube video category currently in Nigeria!


NigeriaYoutubeData = pd.read_csv('NG_youtube_data.csv')

In [40]:
df = NigeriaYoutubeData
df.sort_values(by=['VideoCategoryDescription','viewCount', 'title', 'Description']).head(50)

title      videoId  \
14                              The Reign of Omo Igbo  ASnTvDizlWY   
49                                  Police Internship  veIB8gLx5GM   
0   FAMILY MATTERS | Brodashaggi | Mr Ibu | Kiriku...  Zv1i-Cdn4Ik   
18                      Calling the Police Gone Wrong  Ol2y-pn_XrA   
35                                      SEE WAHALA O   qo-K3YlcfU0   
40  PERFECT ASSISTANT- Watch Maurice Sam and Shine...  MyVG-UcdLU8   
17  Aranda Ekun Part 3 - Latest Yoruba Movie 2022 ...  bC95DBzHuw8   
25  Mr & Mrs Sunday 2 Yoruba Latest Movie 2022 Dra...  jRYcRLkh4R8   
24  Olosho Ghetto 3 Latest Yoruba Movie 2022 Drama...  ROEYx5grjZ8   
16  ALANI OGO - Latest 2022 Yoruba Movie Drama Sta...  _Q8nPeKwYYk   
47                             Akpan and Oduma 'JAPA'  2C1DTt1JFUY   
39  A Dying Wish Yoruba Latest Movies 2022 Drama |...  BJvsOilWQUU   
8   IYUNOLA(AGBOLUAJE) 2 -Latest 2022 Yoruba Movie...  8ysbPIw76i4   
41  THE NEW WIFE MUST GO | MR MACARONI | TAAOOMA |...  tgjVBObnW6E   
4                     IZZAR SO SABON SALO EPISODE 102  lywE4C4ewmk   
2   ROMOLA ALAKARA - Latest 2022 Yoruba Movie Star...  seZb9P7H6MU   
30           IZZAR SO SABON SALO EPISODE 101 ORIGINAL  6WVcKaFS8Xk   
10  KAFAYA FAYA - Mercy Johnson and Ebube Nwagbo B...  tFEfCqKu58o   
19  Olosho Ghetto 2 Latest Yoruba Movie 2022 Drama...  D14mXmRjWms   
33    tHDyr lfTr wwl tjrbh llfwl lm`lb - mTbkh khdyj@  MCC4k5Kn3zE   
46                   The Rock Vs MrBeast For $100,000  S_CUEOBZ0P4   
43  OKO (Stone) Latest Yoruba Movie 2022 Mide Mart...  Poe-E4flB2I   
31  Street Alhaja - A Yoruba Movie Starring Bimpe ...  FcOcDbzxDPA   
1   Best Friends in the World: Senior Year | Episo...  BmCkK7W7j0Q   
7   Best Friends in the World: Senior Year | Episo...  taSmzW5wvZM   
26  Blaqbonez - Fake Nikes (Feat. Blxckie & Cheque...  oL_NxgHJ_M4   
45    Phyno - BBO (Bad Bxtches Only) [Official Video]  TCchFx_dgp8   
27            Bella Shmurda - Contraband (Visualizer)  b5XJf8uQ-Xc   
3                                        Money & Love  pbATe_uZFTE   
12                                               Rush   Xd1VSwCi2o   
23       Bella Shmurda & Simi - Loose it (Visualizer)  Hs6GIQyAR5g   
29  Lift Me Up (From Black Panther: Wakanda Foreve...  Y6p5FLuv-7Y   
38  Kizz Daniel, EMPIRE - Cough (Official Lyric Vi...  D4wfz73Nqw4   
15                                        Cough (Odo)  VLDOAM-8K8E   
28  Fireboy DML - All Of Us (Ashawo) (Official Video)  FfHSi48IY4g   
32                 Oxlade - KU LO SA (Official Video)  R4qud199tQk   
9                 Wizkid - Bad To Me (Official Video)  2mkhevOXVNk   
44                               Rihanna - Lift Me Up  Mx_OexsUI2M   
11     Blaqbonez - Back In Uni (Official Music Video)  s4y_kzpCthQ   
20                  Rihanna - Lift Me Up (Visualizer)  ospQ06jJe-I   
21  Chris Brown - Under The Influence (Official Vi...  pfxyk1glEq4   
36  (SEE VIDEO) President Buhari's "Look Alike" En...  JFHKNz2TL0c   
6   Peter Obi to Presidential Candidates: Tell Nig...  sBX_9gpyqKg   
48                  Kizz daniel BUGA Wedding Entrance  oOzKbiWWl20   
42  EVERTON 3-0 CRYSTAL PALACE | Premier League hi...  rND-d_m19Dg   
5   Roma-Napoli 0-1 | Osimhen stuns Roma with a be...  Ga_i97g0uFY   
22  A FIRST Goal For Casemiro  | Chelsea 1-1 Man U...  klG_itC5ROw   
37       Resumen de Real Madrid vs FC Barcelona (3-1)  swLI6zuXtp4   
13  UNBELIEVABLE Performance!  | Man Utd 2-0 Totte...  O0W4e7d4VvE   
34  HIGHLIGHTS: Liverpool 1-0 Manchester City | Sa...  TE0e64xbtqQ   

                   channelId             channelTitle  categoryId  \
14  UCIwUNuYEcZiiJqjU04yN30A  AyoAjewole Woliagba-YPM          23   
49  UC26q7KQxr5TnKm-OT25MyyQ             HouseOfAjebo          23   
0   UCG6orNVuXIICv9_ifH6msIA    official BRODA SHAGGI          23   
18  UCEZ6QFA7F37E--lqew7wbAQ      Ofentse Mwase Films          23   
35  UCjSzBGfo9gDXP0OerKJ9GZg              Twyse Ereme          24   
40  UCnZ28GESUBXyZSIvzcs1RXA        ChinneyLoveEze Tv          

In [49]:
df.sort_values(by=['VideoCategoryDescription','viewCount', 'title', 'Description'], ascending=(True,False,False,False)).head(50)

title      videoId  \
18                      Calling the Police Gone Wrong  Ol2y-pn_XrA   
0   FAMILY MATTERS | Brodashaggi | Mr Ibu | Kiriku...  Zv1i-Cdn4Ik   
49                                  Police Internship  veIB8gLx5GM   
14                              The Reign of Omo Igbo  ASnTvDizlWY   
46                   The Rock Vs MrBeast For $100,000  S_CUEOBZ0P4   
33    tHDyr lfTr wwl tjrbh llfwl lm`lb - mTbkh khdyj@  MCC4k5Kn3zE   
19  Olosho Ghetto 2 Latest Yoruba Movie 2022 Drama...  D14mXmRjWms   
10  KAFAYA FAYA - Mercy Johnson and Ebube Nwagbo B...  tFEfCqKu58o   
30           IZZAR SO SABON SALO EPISODE 101 ORIGINAL  6WVcKaFS8Xk   
2   ROMOLA ALAKARA - Latest 2022 Yoruba Movie Star...  seZb9P7H6MU   
4                     IZZAR SO SABON SALO EPISODE 102  lywE4C4ewmk   
41  THE NEW WIFE MUST GO | MR MACARONI | TAAOOMA |...  tgjVBObnW6E   
8   IYUNOLA(AGBOLUAJE) 2 -Latest 2022 Yoruba Movie...  8ysbPIw76i4   
39  A Dying Wish Yoruba Latest Movies 2022 Drama |...  BJvsOilWQUU   
47                             Akpan and Oduma 'JAPA'  2C1DTt1JFUY   
16  ALANI OGO - Latest 2022 Yoruba Movie Drama Sta...  _Q8nPeKwYYk   
24  Olosho Ghetto 3 Latest Yoruba Movie 2022 Drama...  ROEYx5grjZ8   
25  Mr & Mrs Sunday 2 Yoruba Latest Movie 2022 Dra...  jRYcRLkh4R8   
17  Aranda Ekun Part 3 - Latest Yoruba Movie 2022 ...  bC95DBzHuw8   
40  PERFECT ASSISTANT- Watch Maurice Sam and Shine...  MyVG-UcdLU8   
35                                      SEE WAHALA O   qo-K3YlcfU0   
7   Best Friends in the World: Senior Year | Episo...  taSmzW5wvZM   
1   Best Friends in the World: Senior Year | Episo...  BmCkK7W7j0Q   
31  Street Alhaja - A Yoruba Movie Starring Bimpe ...  FcOcDbzxDPA   
43  OKO (Stone) Latest Yoruba Movie 2022 Mide Mart...  Poe-E4flB2I   
21  Chris Brown - Under The Influence (Official Vi...  pfxyk1glEq4   
20                  Rihanna - Lift Me Up (Visualizer)  ospQ06jJe-I   
11     Blaqbonez - Back In Uni (Official Music Video)  s4y_kzpCthQ   
44                               Rihanna - Lift Me Up  Mx_OexsUI2M   
9                 Wizkid - Bad To Me (Official Video)  2mkhevOXVNk   
32                 Oxlade - KU LO SA (Official Video)  R4qud199tQk   
28  Fireboy DML - All Of Us (Ashawo) (Official Video)  FfHSi48IY4g   
15                                        Cough (Odo)  VLDOAM-8K8E   
38  Kizz Daniel, EMPIRE - Cough (Official Lyric Vi...  D4wfz73Nqw4   
29  Lift Me Up (From Black Panther: Wakanda Foreve...  Y6p5FLuv-7Y   
23       Bella Shmurda & Simi - Loose it (Visualizer)  Hs6GIQyAR5g   
12                                               Rush   Xd1VSwCi2o   
3                                        Money & Love  pbATe_uZFTE   
27            Bella Shmurda - Contraband (Visualizer)  b5XJf8uQ-Xc   
45    Phyno - BBO (Bad Bxtches Only) [Official Video]  TCchFx_dgp8   
26  Blaqbonez - Fake Nikes (Feat. Blxckie & Cheque...  oL_NxgHJ_M4   
6   Peter Obi to Presidential Candidates: Tell Nig...  sBX_9gpyqKg   
36  (SEE VIDEO) President Buhari's "Look Alike" En...  JFHKNz2TL0c   
48                  Kizz daniel BUGA Wedding Entrance  oOzKbiWWl20   
34  HIGHLIGHTS: Liverpool 1-0 Manchester City | Sa...  TE0e64xbtqQ   
13  UNBELIEVABLE Performance!  | Man Utd 2-0 Totte...  O0W4e7d4VvE   
37       Resumen de Real Madrid vs FC Barcelona (3-1)  swLI6zuXtp4   
22  A FIRST Goal For Casemiro  | Chelsea 1-1 Man U...  klG_itC5ROw   
5   Roma-Napoli 0-1 | Osimhen stuns Roma with a be...  Ga_i97g0uFY   
42  EVERTON 3-0 CRYSTAL PALACE | Premier League hi...  rND-d_m19Dg   

                   channelId             channelTitle  categoryId  \
18  UCEZ6QFA7F37E--lqew7wbAQ      Ofentse Mwase Films          23   
0   UCG6orNVuXIICv9_ifH6msIA    official BRODA SHAGGI          23   
49  UC26q7KQxr5TnKm-OT25MyyQ             HouseOfAjebo          23   
14  UCIwUNuYEcZiiJqjU04yN30A  AyoAjewole Woliagba-YPM          23   
46  UCX6OQ3DkcsbYNE6H8uQQuVA                  MrBeast          24   
33  UCSw3TBd9ohpgHzq1dPBKNsA          Khadija channel          

# Sorting Needed Columns for United State's Current Trending Youtube Video Dataset:
###### Here we are creating a dataframe that shows the trending United State Dataset by Video Category Label and View Count in ascending order.

###### So we have 13 trending United State YouTube Video Categories and they are as follows:
       Comedy, Entertainment, Film & Animation, Music, News & Politics, People & Blogs, Sports, Education, Science & Technology, Howto & Style, Nonprofits & Activism, Gaming, and Travel & Events.
    
###### The highest trending video under Comedy has 515 viewCounts
###### The highest trending video under <font color=red>Entertainment</font> has 148,681 viewCounts
###### The highest trending video under Film & Animation has 4,040 viewCounts
###### The highest trending video under Music has 160 viewCounts 
###### The highest trending video under News & Politics has 108,974 viewCounts
###### The highest trending video under People & Blogs has 18,908 viewCounts
###### The highest trending video under Sports has 1,805 viewCounts
###### The highest trending video under  Education has 35,928 viewCounts
###### The highest trending video under Science & Technology has 30 viewCounts
###### The highest trending video under  Howto & Style has 294 viewCounts
###### The highest trending video under  Nonprofits & Activism has 1,512 viewCounts
###### The highest trending video under Gaming has 1,851 viewCounts
###### The highest trending video under Travel & Events has 843 viewCounts


###### Out of all the 13 trending United State Youtube Video Categories, <font color=red>Entertainment</font> again seems to have the most viewcount , therefore making it the highest trending youtube video category currently in the United States!



In [50]:
USYoutubeData = pd.read_csv('us_Youtube_data.csv')


In [51]:
df = USYoutubeData
df.sort_values(by=['videoCategoryLabel','viewCount', 'videoTitle', 'videoDescription']).head(115)


Unnamed: 0                 channelId       channelTitle      videoId  \
61           61  UCZKcHstffFwoxnTs5uKmryw    kodie lallement  P1Zb1ixhesA   
65           65  UCg2EubjIryo4KZnrhIK7LeA          InsideTRU  xijwKjyY3JI   
105         105  UCZA4UoKQSLUzJ3g4DLZS4EA    wemakeideasflow  XqLro_GPPIc   
64           64  UCfBmmf1OpYLL7JrD2ExMQLA   VIZAKIS MODALITY  tvtzuL45dbc   
114         114  UCH4ezCqRXHQhsETfl0dainA  FabFormIndustries  ODX4tzCWhhU   
..          ...                       ...                ...          ...   
35           35  UCKa6yh0Vd3-NNs64Ea5dihg        CapsugelPTG  bPxSV0AU5qc   
32           32  UCKa6yh0Vd3-NNs64Ea5dihg        CapsugelPTG  XuRmdIib9V0   
9             9  UCn8V3KNSgDr1Dai77_y8JrQ   PsychedSubstance  UGxSZqTZVaU   
48           48  UCnY0P7Z7FBMMIoHC2zUJVIQ         Nutristore  hF3mEHe3klk   
36           36  UC7wOvqC3gyy45Eevq-ZSokw           jayurbzz  GpCpDsPuZO4   

                  publishedAt  \
61   2014-01-18T12:10:49.000Z   
65   2015-05-27T19:37:09.000Z   
105  2013-04-11T17:04:03.000Z   
64   2016-04-11T05:16:48.000Z   
114  2015-08-19T18:25:20.000Z   
..                        ...   
35   2011-05-27T09:23:31.000Z   
32   2011-05-27T09:10:20.000Z   
9    2015-09-12T13:44:39.000Z   
48   2014-10-25T02:52:13.000Z   
36   2016-03-02T21:05:30.000Z   

                                            videoTitle  \
61               con cret micro dosing creatine review   
65   3MT - Erika Bachmann: Fertilizer microdosing f...   
105                           7615 Micro Dosing Pump-1   
64        Psychedelic Micro dosing and Weight Training   
114                           Dosing Helix Micro Rebar   
..                                                 ...   
35   Xcelodose 600S Precision Powder Micro-dosing S...   
32   Xcelodose 120S Precision Powder Micro-dosing S...   
9                   Microdosing LSD or any Psychedelic   
48   CON-CRET CREATINE MICRO-DOSING 48 Servicios PR...   
36   Microdosing LSD (2/5): Staying Present in Real...   

                                      videoDescription  videoCategoryId  \
61   my opinion and review and concret micro dosing...               23   
65   Erika Bachmann, Agricultural Economics, Univer...               27   
105                                                                  27   
64   Join me as I explore the world of micro dosing...               27   
114  Coquitlam concrete goes into detail how they d...               27   
..                                                 ...              ...   
35   Fully automated filling of pharmaceutical powd...               28   
32   Semi-automatic filling of pharmaceutical powde...               28   
9    How to microdose psychedelics! Everything you ...               28   
48   CON-CRET CREATINE MICRO-DOSING 48 Servicios PR...               17   
36   For educational purposes: what to (kind of) ex...               19   

       videoCategoryLabel  duration  durationSec definition  caption  \
61                 Comedy   PT3M10S          190         sd    False   
65              Education   PT2M52S          172         hd    False   
105             Education     PT43S           43         sd    False   
64              Education   PT2M53S          173         hd    False   
114             Education      PT1M            1         hd    False   
..                    ...       ...          ...        ...      ...   
35   Science & Technology   PT7M14S          434         sd    False   
32   Science & Technology   PT7M26S          446         sd    False   
9    Science & Technology  PT13M25S          805         hd    False   
48                 Sports   PT4M34S          274         hd    False   
36        Travel & Events   PT6M17S          377         hd    False   

     viewCount  likeCount  dislikeCount  commentCount  
61         515        2.0           0.0           0.0  
65          38        0.0           0.0           0.0  
105        218        0.0           0.0           0.

In [61]:
df.sort_values(by=['videoCategoryLabel','viewCount', 'videoTitle', 'videoDescription'], ascending=(True,False,False,False)).head(115)


Unnamed: 0                 channelId           channelTitle      videoId  \
61          61  UCZKcHstffFwoxnTs5uKmryw        kodie lallement  P1Zb1ixhesA   
33          33  UCXSF1F_RFRUVNXX4QmB6vmw          Your Mate Tom  iG6nxxfx42w   
10          10  UCXSF1F_RFRUVNXX4QmB6vmw          Your Mate Tom  cnFTo2zFUgw   
31          31  UCXSF1F_RFRUVNXX4QmB6vmw          Your Mate Tom  kbVwK9hqSKc   
44          44  UCWhd7Voh_8ZjyM2NryqpMbg   TheOriginsOfReligion  waKpmL2rfaQ   
..         ...                       ...                    ...          ...   
17          17  UCmSvUqDxMhu2CN5PqkTK1Eg              CABIvideo  wgjXgfzqvW0   
16          16  UCmSvUqDxMhu2CN5PqkTK1Eg              CABIvideo  xl7pRQGIMdg   
50          50  UCxw4-gbOtqRLdh9C3Ne-XAw  E-PAK Machinery, Inc.  b0RDMUgn_3E   
48          48  UCnY0P7Z7FBMMIoHC2zUJVIQ             Nutristore  hF3mEHe3klk   
36          36  UC7wOvqC3gyy45Eevq-ZSokw               jayurbzz  GpCpDsPuZO4   

                 publishedAt  \
61  2014-01-18T12:10:49.000Z   
33  2015-11-10T09:20:00.000Z   
10  2015-12-05T16:26:46.000Z   
31  2015-11-13T14:35:06.000Z   
44  2015-10-12T23:19:55.000Z   
..                       ...   
17  2013-05-17T12:33:05.000Z   
16  2013-04-29T05:35:47.000Z   
50  2016-06-03T20:28:37.000Z   
48  2014-10-25T02:52:13.000Z   
36  2016-03-02T21:05:30.000Z   

                                           videoTitle  \
61              con cret micro dosing creatine review   
33  Microdosing PSILOCYBIN Mushrooms Experiment | ...   
10  Microdosing PSILOCYBIN Mushrooms Experiment ||...   
31          Microdosing PSILOCYBIN Experiment | Day 2   
44  Microdosing Psilocybe Testimony - The Natural ...   
..                                                ...   
17                                        Microdosing   
16                       Fighting striga-Micro-dosing   
50  Fully Automatic Micro Dosing Piston Filler for...   
48  CON-CRET CREATINE MICRO-DOSING 48 Servicios PR...   
36  Microdosing LSD (2/5): Staying Present in Real...   

                                     videoDescription  videoCategoryId  \
61  my opinion and review and concret micro dosing...               23   
33  Support me on Patreon: https://www.patreon.com...               27   
10  I DO NOT CONDONE THE USE OF ANY DRUGS --------...               27   
31  Microdosing PSILOCYBIN | Day 1 https://www.you...               27   
44  Microdosing Psilocybe Testimony - The Natural ...               27   
..                                                ...              ...   
17                    Farmer to farmer training video               28   
16  Farmer to farmer training video on fighting st...               28   
50  8 Head Fully Automatic Micro Dosing Piston Fil...               28   
48  CON-CRET CREATINE MICRO-DOSING 48 Servicios PR...               17   
36  For educational purposes: what to (kind of) ex...               19   

      videoCategoryLabel  duration  durationSec definition  caption  \
61                Comedy   PT3M10S          190         sd    False   
33             Education   PT7M24S          444         hd    False   
10             Education  PT12M32S          752         hd    False   
31             Education   PT7M32S          452         hd    False   
44             Education   PT4M55S          295         hd    False   
..                   ...       ...          ...        ...      ...   
17  Science & Technology   PT9M41S          581         sd    False   
16  Science & Technology   PT9M41S          581         sd    False   
50  Science & Technology   PT4M17S          257         hd    False   
48                Sports   PT4M34S          274         hd    False   
36       Travel & Events   PT6M17S          377         hd    False   

    viewCount  likeCount  dislikeCount  commentCount  
61        515        2.0           0.0           0.0  
33      35928      323.0          12.0         178.0  
10       8869      298.0           1.0          80.0  
31       58

In [62]:
df.sort_values(by=['videoCategoryLabel','viewCount', 'videoTitle', 'videoDescription'], ascending=(True,False,False,False)).head(60)


Unnamed: 0                 channelId                   channelTitle  \
61           61  UCZKcHstffFwoxnTs5uKmryw                kodie lallement   
33           33  UCXSF1F_RFRUVNXX4QmB6vmw                  Your Mate Tom   
10           10  UCXSF1F_RFRUVNXX4QmB6vmw                  Your Mate Tom   
31           31  UCXSF1F_RFRUVNXX4QmB6vmw                  Your Mate Tom   
44           44  UCWhd7Voh_8ZjyM2NryqpMbg           TheOriginsOfReligion   
114         114  UCH4ezCqRXHQhsETfl0dainA              FabFormIndustries   
64           64  UCfBmmf1OpYLL7JrD2ExMQLA               VIZAKIS MODALITY   
105         105  UCZA4UoKQSLUzJ3g4DLZS4EA                wemakeideasflow   
65           65  UCg2EubjIryo4KZnrhIK7LeA                      InsideTRU   
29           29  UCn8V3KNSgDr1Dai77_y8JrQ               PsychedSubstance   
30           30  UC6kFuQCLBFJnIzzhC5cysaA     The Natural Way of Healing   
5             5  UCxSbJSgLqIO8r7TONBN5dvw       Oracle Frequency Podcast   
4             4  UCFmLi6X1mojkFZOFngNR9tQ          Drug Education Agency   
13           13  UCWdEQ_cOzB1S3MEGHIBnxSA                       Nate God   
22           22  UC6kFuQCLBFJnIzzhC5cysaA     The Natural Way of Healing   
113         113  UC281yIVp-U8Ijg4KUVRgDRA                       Osiron X   
110         110  UCjnXuO6mfE0-czvcbOYnpbQ                Ayahuasca Magic   
88           88  UCFlKVHW74SjZrmM4KcIu_kg                    Sabina Cruz   
3             3  UCgbWWPn3VYYzxjffZbfj9GQ                Alan Springwind   
39           39  UCoM5UvXoqDHd2zCpqTrUm3w                   Malan Darras   
59           59  UC9B2G9dYd4dIbQtyy9SczAQ                  David Zobrist   
111         111  UC0dsjSDsLZ5hbmoNjvsHJsQ                  fluidresearch   
28           28  UCBreA_KjlvTPX0P3u2Fxn3g                    Edgar Frias   
21           21  UCewQJkBg5UT2Q2qXNmeUmHQ                      MicroLife   
7             7  UCmrLCXSDScliR7q8AxxjvXg            Black Pigeon Speaks   
23           23  UCldfgbzNILYZA4dmDt4Cd6A                   Secular Talk   
37           37  UCvb-DqrtgHRX-hPypnjfkdw   The Point with Ana Kasparian   
41           41  UCuUJ0ByfT_9idrRzIFu9OyA                      TheLipTV2   
67           67  UCFC0cHr4iU73zuQVMKtSHKw                     Agrivideos   
53           53  UCF4LEacx9sDvAbEJnlaJfKQ                CBS SF Bay Area   
0             0  UCU1_l0ZJyTK_7HZZ3Ruw8Dg                           MAPS   
19           19  UC7wOvqC3gyy45Eevq-ZSokw                       jayurbzz   
66           66  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
42           42  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
18           18  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
84           84  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
60           60  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
106         106  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
72           72  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
87           87  UCfvKwp-BExobsDcuDfCx1hA                     ICRISAT Co   
27           27  UC-MsM6mCPLnUrjOAfsCMTxw                 Young Morpheus   
12           12  UCpy50X_0LZU3Ju2GTRZ6l7w              Vanrooy Machinery   
14           14  UCpy50X_0LZU3Ju2GTRZ6l7w              Vanrooy Machinery   
38           38  UC6kFuQCLBFJnIzzhC5cysaA     The Natural Way of Healing   
75           75  UCI-jLIlZKXHCZApl3VvWfgQ            The Awakened Brave!   
107         107  UC3M1LDs2tSg0Gy3iCqMLGRg              Nöron Mühendislik   
43           43  UCyPSHKvZDBwmGtQxaF9Hg7A                    trippyshore   
1             1  UCLuO2lUqHrPIIpx0hFenV2g                 Tink Tink Club   
6             6  UCSqJniChpE6ngCRzUvNNgug            Fabian-ji Chononita   
77           77  UCI-jLIlZKXHCZApl3VvWfgQ            The Awakened Brave!   
82           82  UCmReYKJm8ohBE4CgiyBMvQA                   VideoOSVTech   
109         109  UCI-jLIlZKXHCZApl3VvWfgQ            